dask_glm/estimators.py

```diff
@@ -0,0 +1,144 @@
+"""
+Models following scikit-learn's estimator API.
+"""
+from . import algorithms
+from . import families
+from . import regularizers
+from .utils import sigmoid, dot, add_intercept
+
+
+class _BaseEstimator(object):
+
+    @property
+    def family(self):
+        """
+        Family
+        """
+
+    def __init__(self, X=None, y=None,
+                 fit_intercept=True, solver='admm', regularizer=None, max_iter=100, tol=1e-4,
+                 lamduh=None, rho=None, over_relax=None, abstol=None, reltol=None):
+        # Deferring the API discussion, we accept both scikit-learn style and
+        # statsmodels style for now. Will most likely change in the future.
+        if (X is None) is not (y is None):
+            raise TypeError("`X` and `y` must both be None, or not None")
+        self.X = X
+        self.y = y
+
+        self.fit_intercept = True
+        self.solver = algorithms._solvers[solver]
+        if solver in ('admm', 'proximal_grad'):
+            regularizer = regularizer or 'L1'
+            lamduh = lamduh if lamduh is not None else 0.1
+        if solver == 'admm':
+            rho = rho if rho is not None else 1
+            over_relax = over_relax if over_relax is not None else 1
+            abstol = abstol if abstol is not None else 1e-4
+            reltol = reltol if reltol is not None else 1e-2
+
+        self.regularizer = regularizers._regularizers.get(regularizer, None)
+        self.max_iter = max_iter
+        self.tol = tol
+        self.lamduh = lamduh
+        self.rho = rho
+        self.over_relax = over_relax
+        self.abstol = abstol
+        self.reltol = reltol
+        self._validate()
+        self.coef_ = None
+        self.intercept_ = None
+        self._coef = None  # coef, maybe with intercept
+
+        fit_kwargs = {'max_steps', 'tol', 'family'}
+        # TODO: use scikit-learn names throughout the lib
+        renamer = {'max_steps': 'max_iter', 'reg': 'regularizer'}
```

Comment:
I think it would also be reasonable to change the existing algorithms to use these names consistently.  I wouldn't be shy about touching that code.